pkg/rpcclient/nep24/royalty.go

```diff
@@ -0,0 +1,182 @@
+// Package nep24 provides RPC wrappers for NEP-24 contracts, which define NFT royalties.
+package nep24
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/nspcc-dev/neo-go/pkg/neorpc/result"
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient/nep11"
+	"github.com/nspcc-dev/neo-go/pkg/rpcclient/neptoken"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+	"github.com/nspcc-dev/neo-go/pkg/vm/stackitem"
+)
+
+// RoyaltyRecipient contains information about the recipient and the royalty amount.
+type RoyaltyRecipient struct {
+	Address util.Uint160
+	Amount  *big.Int
+}
+
+// RoyaltiesTransferredEvent represents a RoyaltiesTransferred event as defined in the NEP-24 standard.
+type RoyaltiesTransferredEvent struct {
+	RoyaltyToken     util.Uint160
+	RoyaltyRecipient util.Uint160
+	Buyer            util.Uint160
+	TokenID          []byte
+	Amount           *big.Int
+}
+
+// RoyaltyReader is an interface for contracts implementing NEP-24 royalties.
+type RoyaltyReader struct {
+	neptoken.Base
+	Invoker nep11.Invoker
+	hash    util.Uint160
+}
+
+// Royalty is a full reader interface for NEP-24 royalties.
+type Royalty struct {
+	RoyaltyReader
+}
+
+// NewRoyaltyReader creates an instance of RoyaltyReader for a contract with the given hash using the given invoker.
+func NewRoyaltyReader(invoker nep11.Invoker, hash util.Uint160) *RoyaltyReader {
+	return &RoyaltyReader{
+		Base:    *neptoken.New(invoker, hash),
+		Invoker: invoker,
+		hash:    hash,
+	}
+}
+
+// NewRoyalty creates an instance of Royalty for a contract with the given hash using the given actor.
+func NewRoyalty(actor nep11.Actor, hash util.Uint160) *Royalty {
+	return &Royalty{
+		RoyaltyReader: *NewRoyaltyReader(actor, hash),
+	}
+}
+
+func (c *RoyaltyReader) RoyaltyInfo(tokenID []byte, royaltyToken util.Uint160, salePrice *big.Int) ([]RoyaltyRecipient, error) {
+	res, err := c.Invoker.Call(c.hash, "royaltyInfo", tokenID, royaltyToken, salePrice)
+	if err != nil {
+		return nil, err
+	}
+	var royalties []RoyaltyRecipient
+	for _, item := range res.Stack {
+		royalty, ok := item.Value().([]stackitem.Item)
+		if !ok || len(royalty) != 2 {
```

Comment:
Len check doesn't belong here.

Reply:
Without a fix or a comment, how are we supposed to interact?

Reply:
I did not change it because the whole implementation of `func (c *RoyaltyReader) RoyaltyInfo(tokenID []byte, royaltyToken util.Uint160, salePrice *big.Int) ([]RoyaltyRecipient, error) {` is questionable and should align with auto-generated, but now it doesn't and is handwritten based on examples. So the whole function will be changed.

Reply:
Let it be handwritten, we need a proper code for `nep24` package. Actually, both `nep11` and `nep17` are partially-handwritten.